diff --git a/extension/extension_build_tools.cmake b/extension/extension_build_tools.cmake
index 977a4b2e36..7cf6d5036e 100644
--- a/extension/extension_build_tools.cmake
+++ b/extension/extension_build_tools.cmake
@@ -193,7 +193,7 @@ function(build_loadable_extension_directory NAME ABI_TYPE OUTPUT_DIRECTORY EXTEN
         add_custom_command(
                 TARGET ${TARGET_NAME}
                 POST_BUILD
-                COMMAND emcc $<TARGET_FILE:${TARGET_NAME}> -o $<TARGET_FILE:${TARGET_NAME}>.wasm -O3 -sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS="${EXPORTED_FUNCTIONS}" ${WASM_THREAD_FLAGS} ${TO_BE_LINKED}
+                COMMAND emcc $<TARGET_FILE:${TARGET_NAME}> -o $<TARGET_FILE:${TARGET_NAME}>.wasm -O1 -sSIDE_MODULE=2 -sEXPORTED_FUNCTIONS="${EXPORTED_FUNCTIONS}" ${WASM_THREAD_FLAGS} ${TO_BE_LINKED}
         )
     endif()
 
